pkg/upgrade/upgrade.go

```diff
@@ -329,6 +339,44 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform, appNS
 	return nil
 }
 
+func CleanupExistingResource(cli client.Client, platform deploy.Platform) error {
+	var multiErr *multierror.Error
+
+	// Special Handling of cleanup of deprecated model monitoring stack
+	if platform == deploy.ManagedRhods {
```

Comment:
if we want to add `rhods-monitor-federation2` in the function, we will need it to be supported for both Self and Managed type

Reply:
we can add a line outside the conditional block for ManagedRhods 
`multiErr = multierror.Append(multiErr, deleteDeprecatedServiceMonitors(ctx, cli, montNamespace, []string{"rhods-monitor-federation2"}))`